### PR TITLE
Handle disposed TestRunRequest object in Abort.

### PR DESCRIPTION
**Pull request by Faizan2304**
Resolves: https://github.com/Microsoft/vstest/issues/1256

### CODE REVIEW

**Review by smadala — APPROVED** [review]

File: src/Microsoft.TestPlatform.Client/Resources/Resources.resx [hunk 1]
```diff
@@ -117,6 +117,12 @@
   <resheader name="writer">
     <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
   </resheader>
+  <data name="AbortRequested" xml:space="preserve">
```
[the remark this answers is not in the document]
Comment (Faizan2304) [reply src/Microsoft.TestPlatform.Client/Resources/Resources.resx]: Ok..

File: src/Microsoft.TestPlatform.Client/Execution/TestRunRequest.cs [hunk 2]
```diff
@@ -231,12 +223,13 @@ public bool WaitForCompletion(int timeout)
...
         {
             EqtTrace.Verbose("TestRunRequest.CancelAsync: Canceling.");
+            this.LogMessage(ClientResources.CancelationRequested, TestMessageLevel.Informational);
 
             lock (this.syncObject)
             {
                 if (this.disposed)
                 {
-                    throw new ObjectDisposedException("testRunRequest");
+                    EqtTrace.Warning("Ignoring TestRunRequest.CancelAsync() as testRunRequest aboject has already been disposed.");
```
[the remark this answers is not in the document]
Comment (Faizan2304) [reply src/Microsoft.TestPlatform.Client/Execution/TestRunRequest.cs]: Its an implementation details. I am not sure we should notify user.
@codito : input please.

[the remark this answers is not in the document]
Comment (codito) [reply src/Microsoft.TestPlatform.Client/Execution/TestRunRequest.cs]: I think we should retain the `ObjectDisposedException`, it is actually an exceptional scenario. Did we find how the `Abort` can come up for a disposed object?

**Comment by Faizan2304** [comment]
Aborting this PR as we are fixing this [here](https://github.com/Microsoft/vstest/pull/1273)

**Pull request closed without merging**